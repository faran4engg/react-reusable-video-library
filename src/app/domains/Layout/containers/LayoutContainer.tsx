import { useState, FC, useEffect } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import routes from 'app/router/routes';

import { NavSidebar, Content, Header } from 'app/domains/Layout/components';
import { NotFoundPage } from 'app/pages/NotFoundPage/async';

const cls = {
  layout:
    'flex h-screen bg-light-mode-content-bg dark:bg-dark-mode-content-bg text-gray-500 dark:text-gray-200',
  mainWrapper: 'flex flex-col flex-1 w-full',
};

const LayoutContainer: FC = () => {
  const [isSidebarOpen, setIsSidebarOpen] = useState(false);

  useEffect(() => {
    if (!localStorage.twTheme) return;
    document.querySelector('html')!.classList.add(localStorage.twTheme);
  }, []);

  return (
    <div className={cls.layout}>
      <NavSidebar
        isSidebarOpen={isSidebarOpen}
        handleIsSidebarOpen={() => setIsSidebarOpen(prevFlag => !prevFlag)}
      />

      <div className={cls.mainWrapper}>
        <Header
          handleIsSidebarOpen={() => setIsSidebarOpen(prevFlag => !prevFlag)}
        />

        <Content>
          <Switch>
            {routes.map(route =>
              route.component ? (
                <Route
                  key={route.path}
                  exact={true}
                  path={`${route.path}`}
                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
                  render={(props: any) => <route.component {...props} />}
                />
              ) : null,
            )}
            <Redirect exact from="/" to="/home" />

            <Route component={NotFoundPage} />
          </Switch>
        </Content>
      </div>
    </div>
  );
};

export default LayoutContainer;
