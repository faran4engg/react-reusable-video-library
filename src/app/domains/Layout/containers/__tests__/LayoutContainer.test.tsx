import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import LayoutContainer from '..';

const renderer = createRenderer();

describe('<LayoutContainer />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<LayoutContainer>{() => <div />}</LayoutContainer>);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
