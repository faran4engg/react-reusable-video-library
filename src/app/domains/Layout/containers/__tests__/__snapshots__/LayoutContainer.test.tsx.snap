// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutContainer /> should render and match the snapshot 1`] = `
<div
  className="flex h-screen bg-light-mode-content-bg dark:bg-dark-mode-content-bg text-gray-500 dark:text-gray-200"
>
  <NavSidebar
    handleIsSidebarOpen={[Function]}
    isSidebarOpen={false}
  />
  <div
    className="flex flex-col flex-1 w-full"
  >
    <Header
      handleIsSidebarOpen={[Function]}
    />
    <Content>
      <Switch>
        <Route
          exact={true}
          path="/home"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/audios"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/audios/:id"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/upload"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/search"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/search/:name/:id"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/watch/:id"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/about"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/404"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/feedback"
          render={[Function]}
        />
        <Redirect
          exact={true}
          from="/"
          to="/home"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </Content>
  </div>
</div>
`;
